core/src/main/java/io/questdb/cairo/wal/WalWriterMetadata.java

```diff
@@ -79,10 +79,11 @@ public static void syncToMetaFile(
     ) {
         metaMem.jumpTo(0);
         // Size of metadata
-        metaMem.putInt(0);
+        metaMem.skip(Integer.BYTES);
```

Comment:
skipping over size can yield arbitrary size value, which isn't an improvement on a consistent 0. What necessitated this change?

Reply:
You're right... The data may not necessarily be zero-initialized.
The case is where the `_meta` file is being overwritten when a column is marked as "dropped" (int colType flips sign) or a new column is added. These changes are made in-place (overwriting the file) and I wanted to update the column count _last_ to allow for concurrent reads.

Reply:
I've patched it.. It's probably good enough the way it is.
The cleaner approach here would be to have a lock file.

Why this is enough (based on my understanding):
* If columns are marked as "dropped" it doesn't matter.. the uploads can be skipped anyway. As such in the worst case we upload a column we didn't need during a concurrent read.
* New column additions can only happen before any data wal txn is written, so it's fine to temporarily ignore them.

The code change was enough to stop tests from failing.

That said, I'd appreciate your thoughts here.

Reply:
It would be good to set up a memory fence and make those last two writes atomic. I don't know if that's already the case with our current API calls. I don't know how we deal with these and MMAP within Java.

Reply:
sorry, i missed your message, lets regroup when you're back